lib/libc/newlib/include/string.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2023 Intel
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_LIBC_NEWLIB_INCLUDE_STRING_H_
+#define ZEPHYR_LIB_LIBC_NEWLIB_INCLUDE_STRING_H_
+
+#include <stddef.h>
+
+#include_next <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+size_t strnlen(const char *, size_t);
+char  *strtok_r(char *, const char *, char **);
```

Comment:
Does the compiler complain when you add identifier names here and the application has defined `_POSIX_C_SOURCE` so that newlib also declares these functions? If not, adding them to make the compliance checks pass would be nice (the names serve as a limited form of documentation if chosen well).

Reply:
I would like to add feature test macros soon on the POSIX side too, so it's good to be aware that it's in the pipes